apps/oracles/eth-rpc/src/lib.rs

```diff
@@ -144,128 +144,198 @@ type MyProvider = alloy::providers::fillers::FillProvider<
-    let calldata = x.calldata().clone();
-    RequestEthCall::latest(&calldata, &address, id)
-}
-
-fn yn_bnbx_call(provider: MyProvider, id: u64) -> RequestEthCall {
-    let address = address!("0x32c830f5c34122c6afb8ae87aba541b7900a2c5f");
-    let contact = YnETHx::new(address, provider.clone());
-    let shares = U256::from(1_000_000_000_000_000_000_u64);
-    let x = contact.convertToAssets(shares);
-    let calldata = x.calldata().clone();
-    RequestEthCall::latest(&calldata, &address, id)
-}
-
-fn yield_fi_yusd_call(provider: MyProvider, id: u64) -> RequestEthCall {
-    let address = address!("0x19Ebd191f7A24ECE672ba13A302212b5eF7F35cb");
-    let contact = YieldFiyUSD::new(address, provider.clone());
-    let x = contact.exchangeRate();
-    let calldata = x.calldata().clone();
-    RequestEthCall::latest(&calldata, &address, id)
-}
-
-async fn fetch_all(resourses: &[FeedConfig]) -> Result<FetchedDataForFeed> {
-    let mut res = FetchedDataForFeed::new();
+async fn fetch_all(resourses: &[FeedConfig]) -> Result<FetchedDataHashMap> {
+    let mut res = FetchedDataHashMap::new();
     for feed_config in resourses {
-        if let Some(value) = fetch_value_for_feed(feed_config).await? {
-            res.insert(feed_config.feed_id, value);
+        let mut responses = vec![];
+        for contract in &feed_config.arguments.contracts {
+            if let Some(value) = contract.fetch(feed_config.feed_id).await {
+                responses.push(value);
+            }
         }
+        res.insert(
+            feed_config.feed_id,
+            FetchedDataForFeed {
+                contacts: feed_config.arguments.contracts.to_vec(),
+                responses,
+            },
+        );
     }
     Ok(res)
 }
 
-async fn fetch_value_for_feed(
-    feed_config: &FeedConfig,
-) -> Result<Option<ResponseEthCall>, anyhow::Error> {
-    let mut last_value = None;
-    for rpc_url_candidate in &feed_config.arguments {
-        let rpc_url: Url = rpc_url_candidate.as_str().parse()?;
-        let provider = ProviderBuilder::new().connect_http(rpc_url.clone());
-        let eth_call = match feed_config.feed_id {
-            100001 => yn_ethx_call(provider.clone(), 12345 + 998),
-            100002 => yield_fi_yusd_call(provider.clone(), 12345 + 128),
-            100003 => yn_bnbx_call(provider.clone(), 12345 + 129),
-            _ => {
-                break;
+impl Contract {
+    pub fn prepare_request(&self, provider: MyProvider, id: u64) -> Option<RequestEthCall> {
+        match self.method_name.as_str() {
+            "convertToAssets" => Some(self.convert_to_assets(provider.clone(), id)),
+            "exchangeRate" => Some(self.exchange_rate(provider.clone(), id)),
+            _ => None,
+        }
+    }
+
+    pub fn convert_to_assets(&self, provider: MyProvider, id: u64) -> RequestEthCall {
+        let contact = VaultABI::new(self.address, provider.clone());
+        let shares = self.param1.unwrap();
```

Comment:
Can we return an error here instead of unwrap? And also return and error in `prepare_request` and do the same logic with breaking in `fetch` when there is an error.